content/blog/intro-to-gamedev/index.md

```diff
@@ -0,0 +1,560 @@
+
+```ts
+    private CollidesWith(otherSprite: PIXI.Sprite)
+    {
+        // player's rectangle
+        var ab = this.sprite.getBounds();
+
+        // sprite we are checking against
+        var bb = otherSprite.getBounds();
+
+        return  !(ab.x > bb.x + bb.width ||
+                ab.x + ab.width < bb.x ||
+                ab.y + ab.height < bb.y ||
+                ab.y > bb.y + bb.height);
+    }
+```
+
+Since we have a way of checking for 1:1 collisions on `Player`, we'll need to modify the `Update()` method to check if it's colliding against any of all the solid `ActiveEntities`.
+
+Let's add this at the end of the method:
+
+```ts
+for (var i = 0; i < activeEntities.length; i++) {
+  var entity = activeEntities[i]
+  if (entity.solid && this.CollidesWith(entity.sprite)) {
+    GameApp.GameOver = true
+  }
+}
+```
+
+## Finishing touches
+
+We are saving what the max and current score are, but we have not been displaying it so far. Luckily it's pretty easy to such thing in PixiJS. All we need to do is define a [`PIXI.Text`](https://pixijs.download/dev/docs/PIXI.Text.html) object, add it to the stage and adjust it as we need. Let's do it on our `GameApp` class:
+
+```ts
+export class GameApp {
+
+    static ScoreText: PIXI.Text =  new PIXI.Text('Score: ', {fontSize:5 ,fill: '#aaff', align: 'center', stroke: '#aaaaaa', strokeThickness: 0 });
+
+    [....]
+
+    this.app.ticker.add((delta) => {
+            GameApp.Update(delta);
+
+            // if we didn't lose, display score and max score, otherwise show a "game over" prompt
+            if (!GameApp.GameOver)
+            {
+                GameApp.ScoreText.text = "Score: " + (Math.ceil(GameApp.Score)) + " - Max Score: " + Math.ceil(GameApp.MaxScore) ;
+            }
+            else
+            {
+                GameApp.ScoreText.text = "Game over! You scored " + (Math.ceil(GameApp.Score)) +  ". Max Score: " + Math.ceil(GameApp.MaxScore)  + ". Press spacebar to restart.";
+            }
+        });
+}
+```
+
+There are some other finishing touches and optimizations that I added to the code, feel free to review them on the repo (and ask away if you think I can help!).
+
+## Closing
+
+Since it involves many subsystems, there are many, many ways to approach game development in general, and we only just barely scraped the surface of it.
+
+Hopefully this served as a good starting point which you can build upon. TypeScript and PixiJS are both very useful tools that made this very easy, but keep in mind that they are very powerful. There are a number of optimizations and features in terms of code and runtime that could be done and I would suggest exploring to improve your craft: Improve assets loading code by adding compile-time checks (such as for assets names), add different kinds of obstacles, make the character fall faster if the player presses the down arrow ke, recycle/pool objects so that there aren't many allocations, the possibilities are endless!
```

Comment:
it would be awesome to have the game either playable inside this page or a link to a deployed (you can probably use github pages) version (and if you have it deployed perhaps you can embed it within an iframe). I would be a really nice cherry on top of this awesome post.